### PR TITLE
Don't display the expert flow if there are no plans

### PR DESCRIPTION
**Pull request by huwshimi**
This stops the GUI dying if there is an incomplete supported entity.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- use prod with the `expert` flag and visit `/new/?dd=cs:~cmarstech/mattermost`
- the normal deployment flow should appear

**Comment by huwshimi** [comment]
[Fix more gui tables](https://trello.com/c/G5heOxv6/1035-fix-more-gui-tables)

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 1]
```diff
@@ -80,14 +80,15 @@ class DeploymentFlow extends React.Component {
     // If the direct deploy data changes then get and store the new entity or
     // clear the state.
     const isDirectDeploy = !!(nextProps.ddData && nextProps.ddData.id);
-    this.setState({isDirectDeploy});
-    if (isDirectDeploy) {
-      if (nextProps.ddData.id !== this.props.ddData.id) {
-        this._getDirectDeployEntity(nextProps.ddData.id);
+    this.setState({isDirectDeploy}, () => {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Were you seeing race conditions here and below?

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 2]
```diff
@@ -376,6 +379,16 @@ class DeploymentFlow extends React.Component {
     });
   }
 
+  /**
+    Check whether this should be an expert flow.
+
+    @method _isExpertFlow
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: don't need to define the method anymore :)

File: jujugui/static/gui/src/app/components/deployment-flow/expert-intro/expert-intro.js [hunk 3]
```diff
@@ -65,7 +65,8 @@ class DeploymentExpertIntro extends React.Component {
       const entity = entityModel.toEntity();
       const machineNumber = this.isBundle ? entity.machineCount : 1;
       const price = `From $${entity.price} per month`;
-      const supportedDescription = marked(entity.supportedDescription);
+      const supportedDescription = entity.supportedDescription ?
+        marked(entity.supportedDescription) : null;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/expert-intro/expert-intro.js]: We should probably put a message here like "The author has not provided a description." or something along those lines.

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js (hunk 2 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Removed.

File: jujugui/static/gui/src/app/components/deployment-flow/expert-intro/expert-intro.js (hunk 3 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/expert-intro/expert-intro.js]: Done and test added.

**Pull request merged**